### ISSUE TITLE
Does the thread pool need at least 2 threads when profiling is enabled?

### ISSUE THREAD

**Issue by o7si** [google-ai-edge/mediapipe#5734]
### Have I written custom code (as opposed to using a stock example script provided in MediaPipe)

None

### OS Platform and Distribution

macOS 14.5 (Apple M1)

### Programming Language and version

C++

### Bazel version

6.5.0

### Solution

HelloWorld

### Describe the actual behavior

When there is only one Calculator and PROFILING is enabled, the Graph does not execute.

### Describe the expected behaviour

The Graph should execute normally even when there is only one Calculator and PROFILING is enabled.

### Standalone code/steps you may have used to try to get what you need

Detailed steps to reproduce can be found in the main body.

**Comment by o7si** [comment]
**Bug Description**

In the process of using MediaPipe, I found that when only one Calculator is configured in the Graph and the PROFILING function is enabled, the Graph will not be able to process the data stream normally. The specific reason is that since there is only one Calculator configured, the size of the ThreadPool will only be allocated as 1, but the PROFILING function also relies on the thread pool ([and is a long-running loop](https://github.com/google-ai-edge/mediapipe/blob/0cebcc0a5bfe864dc46e2901876c3633367c886e/mediapipe/framework/profiler/graph_profiler.cc#L265)), so when the Graph receives the data stream, there is no thread to schedule the Calculator's Process function. 

**Reproduction Steps**

1. Modify the mediapipe/examples/desktop/hello_world/hello_world.cc file with the following diff:
    ```
    diff --git a/mediapipe/examples/desktop/hello_world/hello_world.cc b/mediapipe/examples/desktop/hello_world/hello_world.cc
    index 50c49d4f..14e779f2 100644
    --- a/mediapipe/examples/desktop/hello_world/hello_world.cc
    +++ b/mediapipe/examples/desktop/hello_world/hello_world.cc
    @@ -26,16 +26,18 @@ absl::Status PrintHelloWorld() {
       // Configures a simple graph, which concatenates 2 PassThroughCalculators.
       CalculatorGraphConfig config =
           ParseTextProtoOrDie<CalculatorGraphConfig>(R"pb(
    +        profiler_config {
    +          trace_enabled: true
    +          enable_profiler: true
    +          trace_log_interval_count: 200
    +          trace_log_path: "./"
    +        }
    +
             input_stream: "in"
             output_stream: "out"
             node {
               calculator: "PassThroughCalculator"
               input_stream: "in"
    -          output_stream: "out1"
    -        }
    -        node {
    -          calculator: "PassThroughCalculator"
    -          input_stream: "out1"
               output_stream: "out"
             }
           )pb");
    ```
2. Build and execute using the following commands:
    ```
    bazel run --define MEDIAPIPE_PROFILING=1 mediapipe/examples/desktop/hello_world:hello_world
    ```

**Observed Behavior**

The following output is observed, indicating that the program failed to execute properly (no threads available for scheduling) as there is no expected “Hello World!” message in the logs:
```
INFO: Analyzed target //mediapipe/examples/desktop/hello_world:hello_world (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //mediapipe/examples/desktop/hello_world:hello_world up-to-date:
  bazel-bin/mediapipe/examples/desktop/hello_world/hello_world
INFO: Elapsed time: 0.097s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/mediapipe/examples/desktop/hello_world/hello_world
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1731757077.208439 32035708 graph_profiler.cc:256] trace_log_path: ./
```

**Comment by o7si** [comment]
Of course, this issue is unlikely to occur under normal circumstances, as projects relying on MediaPipe usually consist of dozens of Calculators, making it highly improbable to have only one. I attempted to fix this issue; see: #5735 for details.

Additionally, I am not particularly familiar with MediaPipe’s implementation, so the root cause I identified might be incorrect. Please pardon any inaccuracies.

**Comment by kuaashish** [comment]
Hi @schmidt-sebastian,

Could you please look into this issue? The relevant PR is available here https://github.com/google-ai-edge/mediapipe/pull/5735.

Thank you!!
